### PR TITLE
feat: MLIBZ-2185 Clear does not clear metadata for data synced in previous sdk versions

### PR DESCRIPTION
**Pull request by yuliya-guseva**
#### Description
`DataStore#clear` does not clear metadata for data synced in previous SDK versions.

#### Changes
Added new method `clearCollection` to CacheManager. The method deletes the collection and creates its new version. Current method `clear` just deletes realm file, this logic was saved. Logic for `DataStore.clear` was updated from `CacheManager#clear` to `CacheManager#clearCollection`.

#### Tests
Instrumented, Manual. Manual testing is necessary in this case because we have to update library version from 3.0.5 to latest version, it's impossible to check in the tests. Anyway, the small test for checking was added, the test checks that `CacheManager#clearCollection` doesn't crash.

### CODE REVIEW

File: android-lib/src/main/java/com/kinvey/android/cache/RealmCacheManager.java [hunk 1]
```diff
@@ -195,6 +195,7 @@ public RealmCacheManager(String prefix, Client client) {
      */
     private void removeSchemas(List<String> tableNames, DynamicRealm realm) {
         RealmSchema realmSchema = realm.getSchema();
+        System.out.println(tableNames);
```
Comment (vinaygahlawat) [inline android-lib/src/main/java/com/kinvey/android/cache/RealmCacheManager.java]: Please remove any debugging statements if not necessary.

File: android-lib/src/main/java/com/kinvey/android/cache/RealmCacheManager.java [hunk 2]
```diff
@@ -211,6 +212,52 @@ public void clear() {
...
+        Set<RealmObjectSchema> schemas = currentSchema.getAll();
+        for (RealmObjectSchema schema : schemas) {
+            className = schema.getClassName();
+            if (className.equals(TableNameManager.getShortName(tableName, realm))) {
+                for (RealmObjectSchema subClassSchema : schemas) {
+                    originalName = TableNameManager.getOriginalName(subClassSchema.getClassName(), realm);
+                    if (originalName == null) {
+                        continue;
+                    }
+                    if (originalName.startsWith(className + "_")) {
```
Comment (vinaygahlawat) [inline android-lib/src/main/java/com/kinvey/android/cache/RealmCacheManager.java]: Should use the underscore constant here if possible.

Comment (yuliya-guseva) [reply android-lib/src/main/java/com/kinvey/android/cache/RealmCacheManager.java]: I agree. Constants.UNDERSCORE was added to another branch (MLIBZ-2392). I'll update it after merging MLIBZ-2392 branch to indev.

**Comment by yuliya-guseva** [comment]
`CacheManager#clearCollection` is used in `DataStore#clear`, because the previous logic of `DataStore#clear` deleted realm file with all tables instead of clearing only selected collection. But this logic when SDK have to delete realm file is used in the `Client#performLockdown` so I just added the new method in the RealmCachemanager(clearCollection) and updated `DataStore#clear`. 
`Client#performLockdown` also is used in the `Logout` request and `Delete(hard)` user, probably  should we use `CacheManager#clearCollection` in this case as well and not delete all data in the realm?

**Comment by vinaygahlawat** [comment]
I think that for the `Logout` case, we still want to delete all the data in Realm.

**Pull request merged**